synthesizer/src/program/mod.rs

```diff
@@ -108,13 +108,28 @@ record credits:
     owner as address.private;
     gates as u64.private;
 
-function mint:
+mapping account:
```

Comment:
I would strongly recommend adding approvals to the default credits program: https://github.com/AleoHQ/aleo/pull/304/files#diff-34ed4dabecf71a6a2db55da718ffbd17767d9c7ec0245300e4da5a348ab0114eR30

Honestly a token standard should probably be designed and worked through before finalizing this program. If it differs from the later implemented token standard, it's going to lead to every program needing to "wrap" microcredits in order for it be used.

Reply:
It may also be time to consider adding `self.parent`. In order for there to be a composable ecosystem of contracts, contracts will need to be able to own assets through mappings.